pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/model/event/CreateKitEventAction.java

```diff
@@ -0,0 +1,129 @@
+        MailAddress defaultAddress = handle.attach(JdbiMailAddress.class)
+                .findDefaultAddressForParticipant(signal.getParticipantGuid())
+                .orElse(null);
+
+        if (defaultAddress == null) {
+            log.warn("Participant {} is missing a default mailing address. Deleting the create kit event", signal.getParticipantGuid());
+            return;
+        }
+
+        if (defaultAddress.getValidationStatus() == DSM_INVALID_ADDRESS_STATUS.getCode()) {
+            log.warn("Participant {} has an invalid mailing address", signal.getParticipantGuid());
+            return;
+        } else {
+            try {
+                statusType = DsmAddressValidationStatus.getByCode(defaultAddress.getValidationStatus());
+            } catch (Exception e) {
+                log.warn(e.getMessage() + ". Participant: {} ", signal.getParticipantGuid());
+                return;
+            }
+        }
+
+        Optional<KitConfigurationDto> kitConfigByTypeOpt = kitConfigs.stream().filter(dto -> dto.getKitTypeId()
+                == kitTypeId).findFirst();
+
+        KitConfiguration kitConfig = kitConfigByTypeOpt.isPresent()
+                ? handle.attach(KitConfigurationDao.class).getKitConfigurationForDto(kitConfigByTypeOpt.get()) : null;
+
+        KitCheckService service = new KitCheckService();
+        KitCheckService.PotentialRecipient candidate = new KitCheckService.PotentialRecipient(user.getId(),
+                user.getGuid(), defaultAddress.getId(), statusType);
+        KitCheckService.KitCheckResult kitCheckResult = new KitCheckService.KitCheckResult();
+        kitCheckResult = service.processPotentialKitRecipient(signal.getStudyGuid(), kitTypeId,
+                kitCheckResult, kitConfig, candidate, true);
+
+        //send metric by study
+        if (kitCheckResult != null) {
+            sendKitMetrics(kitCheckResult);
+            log.info("Queued {} participants for Kit creation", kitCheckResult.getTotalNumberOfParticipantsQueuedForKit());
+        }
+    }
+
+    @VisibleForTesting
+    long queueDelayedEvent(Handle handle, EventSignal signal) {
+        return handle.attach(QueuedEventDao.class).addToQueue(
+                eventConfiguration.getEventConfigurationId(),
+                signal.getOperatorId(),
+                signal.getParticipantId(),
+                eventConfiguration.getPostDelaySeconds());
+    }
+
+    private void sendKitMetrics(KitCheckService.KitCheckResult kitCheckResult) {
+        for (var queuedParticipantsByStudy : kitCheckResult.getQueuedParticipantsByStudy()) {
+            String studyGuid = queuedParticipantsByStudy.getKey();
+            int numQueuedParticipants = queuedParticipantsByStudy.getValue().size();
+            var tracker = kitCounterMonitorByStudy.computeIfAbsent(studyGuid, key ->
+                    new StackdriverMetricsTracker(StackdriverCustomMetric.KITS_REQUESTED, studyGuid,
```

Comment:
There's intermixing of `var` and explicit typing here- can we stick with one or the other?